src/sentry/utils/event_tracker.py

```diff
@@ -0,0 +1,78 @@
+import logging
+from enum import IntEnum
+
+from sentry import options
+
+
+class EventType(IntEnum):
+    TRANSACTION = 1
+    ERROR = 2
+
+
+class TransactionStageStatus(IntEnum):
+    """
+    Do NOT change the int values of existing items. It will change meaning of existing tracking data
+    This class is DELETE or APPEND ONLY
+    """
+
+    # the transaction is stored to rc-transactions
+    REDIS_PUT = 1
+
+    # a save_transactions task is kicked off
+    SAVE_TRX_STARTED = 2
+
+    # a save_transactions task is finished
+    SAVE_TRX_FINISHED = 3
+
+    # the transaction is published to the `events` topic for snuba/sbc consumers to consume
+    SNUBA_TOPIC_PUT = 4
+
+    # the transaction is published to the `snuba-commit-log` topic
+    COMMIT_LOG_TOPIC_PUT = 5
+
+    # a post_process task is kicked off
+    POST_PROCESS_STARTED = 6
+
+    # the transaction is deleted from rc-transactions
+    REDIS_DELETED = 7
+
+
+logger = logging.getLogger("EventTracker")
+
+
+def should_track(event_id: str, sample_rate) -> bool:
```

Comment:
it's wasn't very clear at a glance that this is an internally used function only

```suggestion
def _should_track(event_id: str, sample_rate) -> bool:
```

Reply:
you could probably just inline this into `track_sampled_event`

Reply:
done